File: apps/hello_world_cpp/receiver.cpp
#include "receiver.hpp"
#include <string>

namespace isaac {
    void Receiver::start() {
        tickOnMessage(rx_number());
    }

    void Receiver::tick() {
        auto proto = rx_number().getProto();

        const int number = proto.getMyValue();

        LOG_INFO(std::to_string(number).c_str());
    }
}

File: apps/hello_world_py/codelets/receiver.py
from isaac import Codelet


class Receiver(Codelet):
    def start(self):
        self.input = self.isaac_proto_rx("MyCustomProto", "number")
        self.tick_on_message(self.input)

    def tick(self):
        print(self.input.message.proto.myValue)
